src/diva.js

```diff
@@ -68,18 +85,42 @@ function version() {
 }
 
 // TODO make this more functional
-function addProof(divaSessionState, proof) {
-  divaSessionState.user.attributes.push(proof);
+// TODO merge with existing attributes
+function addAttributesToSession(divaSessionState, attributes) {
+  divaSessionState.user.attributes.push(attributes);
   return divaSessionState;
 }
 
-function deauthenticate() {
-  return {
-    user: {
-      sessionId: uuidv4(),
-      attributes: [],
-    },
-  };
+
+function getProofFromMap(proofMap, sessionId) {
+  if (proofMap.get(sessionId) === undefined) {
+    return BPromise.reject(new Error('Proof does not exist')); // TODO custom error
+  }
+
+  const attributes = proofMap.get(sessionId).attributes;
+
+  return BPromise.resolve(attributes); // TODO also return proof?
+}
+
+function removeSessionFromMap(proofMap, sessionId) {
+  if (pendingProofs.get(sessionId) === undefined) {
+    return BPromise.reject(new Error('Proof does not exist')); // TODO custom error
+  }
+
+  proofMap.delete(sessionId);
```

Comment:
I think this line should be in `getProofFromMap` (which might be better named `getPendingAttributes`